packages/node-app/src/tests/giveaway.controller.test.ts

```diff
@@ -254,3 +256,76 @@ describe('GET /giveaways/:id/generate-winners', () => {
     );
   });
 });
+
+describe('PUT giveaways/:id/participants/:participantId', () => {
+  const participant = {
+    id: 'user',
+    state: ParticipantState.PENDING,
+  };
+
+  const giveawayData = {
+    title: 'Giveaway',
+    description: 'Description for giveaway',
+    startTime: Date.now() + 10000,
+    endTime: Date.now() + 12000,
+    numberOfWinners: 1,
+    prize: 'Test prize',
+    image: 'test-image-base64',
+    participants: [participant],
+  };
+
+  it('should fail if invalid participant', async () => {
+    const giveaway = await Giveaway.create(giveawayData);
+
+    const body = { state: 'invalid' };
```

Comment:
Since this state is also invalid, let's check if the error message is what we expect for this test (invalid participant) or use a valid state to make sure we are testing the right validation